State.py

```diff
@@ -176,12 +180,13 @@ def execute_action(self, player : int, action : Action, ignore_killing : bool =
                 if i + in_hand not in cards_to_keep:
                     self._deck.return_card(card.get_id())
 
-    def validate_action(self, action : Action, player_id : int) -> bool:
+    def validate_action(self, action : Action, player_id : int, whisper: bool = True) -> bool:

```

Comment:
added whisper flag so we can just use this function to truly figure out whether an action is valid